sklearn/compose/tests/test_column_transformer.py

```diff
@@ -794,6 +800,92 @@ def test_column_transformer_no_estimators():
     assert ct.transformers_[-1][2] == [0, 1, 2]
 
 
+@pytest.mark.parametrize(['est', 'pattern'], [
+    (ColumnTransformer(
+        [('trans1', Trans(), [0]),
+         ('trans2', Trans(), [1])],
+        remainder=DoubleTrans()),
+     (r'\[ColumnTransformer\].*\(1 of 3\) Fitting trans1.* elapsed=.*\n'
+      r'\[ColumnTransformer\].*\(2 of 3\) Fitting trans2.* elapsed=.*\n'
+      r'\[ColumnTransformer\].*\(3 of 3\) Fitting remainder.* elapsed=.*\n$')),
+    (ColumnTransformer(
+        [('trans1', Trans(), [0]),
+         ('trans2', Trans(), [1])],
+        remainder='passthrough'),
+     (r'\[ColumnTransformer\].*\(1 of 3\) Fitting trans1.* elapsed=.*\n'
+      r'\[ColumnTransformer\].*\(2 of 3\) Fitting trans2.* elapsed=.*\n'
+      r'\[ColumnTransformer\].*\(3 of 3\) Fitting remainder.* elapsed=.*\n$')),
+    (ColumnTransformer(
+        [('trans1', Trans(), [0]),
+         ('trans2', 'drop', [1])],
+        remainder='passthrough'),
+     (r'\[ColumnTransformer\].*\(1 of 2\) Fitting trans1.* elapsed=.*\n'
+      r'\[ColumnTransformer\].*\(2 of 2\) Fitting remainder.* elapsed=.*\n$')),
+    (ColumnTransformer(
+        [('trans1', Trans(), [0]),
+         ('trans2', 'passthrough', [1])],
+        remainder='passthrough'),
+     (r'\[ColumnTransformer\].*\(1 of 3\) Fitting trans1.* elapsed=.*\n'
+      r'\[ColumnTransformer\].*\(2 of 3\) Fitting trans2.* elapsed=.*\n'
+      r'\[ColumnTransformer\].*\(3 of 3\) Fitting remainder.* elapsed=.*\n$')),
+    (ColumnTransformer(
+        [('trans1', Trans(), [0])],
+        remainder='passthrough'),
+     (r'\[ColumnTransformer\].*\(1 of 2\) Fitting trans1.* elapsed=.*\n'
+      r'\[ColumnTransformer\].*\(2 of 2\) Fitting remainder.* elapsed=.*\n$')),
+    (ColumnTransformer(
+        [('trans1', Trans(), [0]),
+         ('trans2', Trans(), [1])],
+        remainder='drop'),
+     (r'\[ColumnTransformer\].*\(1 of 2\) Fitting trans1.* elapsed=.*\n'
+      r'\[ColumnTransformer\].*\(2 of 2\) Fitting trans2.* elapsed=.*\n$')),
+    (ColumnTransformer(
+        [('trans1', Trans(), [0])],
+        remainder='drop'),
+     (r'\[ColumnTransformer\].*\(1 of 1\) Fitting trans1.* elapsed=.*\n$'))
+])
+@pytest.mark.parametrize('method', ['fit', 'fit_transform'])
+def test_column_transformer_verbose(est, pattern, method, capsys):
+    X_array = np.array([[0, 1, 2], [2, 4, 6], [8, 6, 4]]).T
+
+    func = getattr(est, method)
+    est.set_params(verbose=False)
+    func(X_array)
+    assert not capsys.readouterr()[0], 'Got output for verbose=False'
+
+    est.set_params(verbose=True)
+    func(X_array)
+    assert re.match(pattern, capsys.readouterr()[0])
+
+
+@pytest.mark.parametrize('method', ['fit', 'fit_transform', 'transform'])
+def test_column_transformer_raises_expected_2D_array(method):
```

Comment:
It took me a moment to understand the relevance to this PR. It is because you stopped using FeatureUnion's implementation of fit_transform

Reply:
I updated this PR to use `ColumnTransformer._fit_transform` in `ColumnTransformer.transform`. This should make the test unrelated to this PR, so I removed the test. (Hoping this makes it easier to review)